src/pages/api/preview.js

```diff
@@ -0,0 +1,36 @@
+import wp from "@/promisetracker/lib/wp";
+
+export default async function preview(req, res) {
+  const {
+    postType,
+    postId,
+    revisionId,
+    _wpnonce: nonce,
+    _thumbnail_id: thumbnailId,
+  } = req.query;
+  // Fetch WordPress to check if the provided `id` or `slug` exists
+  const wpApi = wp();
+  let post;
+  if (postType === "pages") {
+    post = await wpApi.revisions({ id: postId, nonce, revisionId, thumbnailId })
+      .page;
+  } else if (postType === "posts") {
+    post = await wpApi.revisions({ id: postId, nonce, revisionId, thumbnailId })
+      .post;
+  }
+  // If the post doesn't exist prevent preview mode from being enabled
+  if (!post) {
+    return res.status(404).redirect(`/not-found`);
+  }
+  res.setPreviewData({
+    query: { id: postId, nonce, revisionId, thumbnailId },
+  });
```

Comment:
Haven't tested it but my understanding of Nextjs is:

> Check if the post we want to preview exist and if it does, take the user to preview page

We currently don't have a method to just check if post exist; meaning we're fetching the full post.

My question is why don't we then pass the full `post` to `setPreviewData` so that in the `getStaticProps` we don't have to fetch it the second time?

Reply:
@kilemensi  there is a limit on what can be passed, and the post exceeds that limit

Reply:
Cool, the why don't skip loading the page here and do everything from `getStaticProps`? We do return 404 from `getStaticProps` for missing pages right?